### DISCUSSION TITLE
Issue to run the CFD simulation in WE-UQ

### DISCUSSION THREAD

**Discussion by Mewtwo1124** [NHERI-SimCenter/SimCenterCommon#304] [WE-UQ]
Hi,
   I tried to run the Wind Load Evaluation on a Generic Isolated Building using CFD but the code stopped and gave the error 

File "C:\Users\Tanmay\Downloads\WE-UQ_Windows_Download_2024-10-09 00-14-25\WE-UQ_Windows_Download\applications\createEVENT\IsolatedBuildingCFD\setup_case.py", line 1989
    <<<<<<< HEAD
    ^
SyntaxError: invalid syntax

The other simulations run just fine but the ones with CFD fail giving the above error while creating the BlockMesh dictionary. I have attached the error file for your reference. Please let me know how to proceed.
![Screenshot 2024-12-06 100235](https://github.com/user-attachments/assets/46ad8361-70dc-4a5e-b0a7-3772e8d371fd)

**Comment by JustinBonus** [comment]
Thanks for your interest in NHERI SimCenter applications!

System / OS information? I'm guessing you're on Windows 10?
 
@abiyfantaye , could you check the IsolatedBuildingCFD/setup_case.py file around line 1989? 

If the bug is not derived from the Python script itself, I'd imagine it has to do with the Python application that is being pointed to in your File >> Preferences in-app, or something to do with the OpenFOAM setup for your system (check the WE-UQ install & run documentation).

**Comment by Mewtwo1124** [comment]
Hi,
  Thanks for the reply. My OS is Windows 11. I downloaded and installed OpenFOAM v10 as mentioned in the documentation and also the test case of 9 story building runs well. Please let me know how to proceed with this.
Thanks
Tanmay

**Reply by abiyfantaye** [reply]
Hi Tanmay,

Thank you for bringing this to our attention and for the detailed information.

We’ve identified the issue as being related to the backend Python script. As a quick fix and to keep you going with your work while we prepare a new release, you  can modify your current setup as follows:

1. Navigate to your installation directory and locate the file: `/applications/createEVENT/IsolatedBuildingCFD/setup_case.py`

2. Edit the file by:

- Removing **line 1249.**

- Removing **lines 1257 to 1262**.

3. Save the file and rerun your model.

This should resolve the issue for now.

Also, we are planning to release an updated version of the WE-UQ application with a minor version update shortly, which will include a permanent fix for this issue. If this quick fix doesn't work, I am available for a Zoom meeting to help debug the issue. Please feel free to reach out, and we can schedule a time.

Thank you for your patience and interest in the tool! Please let us know if you encounter other problems as well.

Cheers,
Abiy

**Comment by Mewtwo1124** [comment]
Hi Abiy,
          I tried the fix but still got the same error message. I have
attached the error message and the setup_case.py file with changes you
suggested. Please let me know how to proceed.

Thanks
Tanmay

[image: image.png]

**Reply by abiyfantaye** [reply]
Hi Tanmay, 

I'm sorry that you are still experiencing the same issue. We have built a new version of the WE-UQ tool with a bug fix. I've tested it on my Windows machine, and it works. Please use the following link and download the latest version: [Download WE-UQ](https://www.designsafe-ci.org/data/browser/tapis/designsafe.storage.community/%2FSimCenter%2FSoftware%2FWE_UQ?_gl=1*104fth8*_ga*MTg0MTMxMTc4MC4xNjgzNzk0NDY0*_ga_D96RT1T24K*MTczMzg4MjgzMi42MDkuMS4xNzMzODgyODg4LjAuMC4w)

Let me know if this fixes the issue. 
Best, 
Abiy

**Reply by Mewtwo1124** [reply]
Hi Abiy,
       I did download the app and when I try to load an example with cfd it gives error saying that there is no controlDict file in the system folder. When I check the system folder there are no files generated. Could you please let me know how can I fix it? I have attached the error file.
Thanks
Tanmay
![Screenshot 2025-01-12 152520](https://github.com/user-attachments/assets/8223a479-e2b6-4c29-b9c2-49c93de79f54)

**Comment by abiyfantaye** [comment]
Hi Tanmay,

In order to assist you more effectively, could you please share the setup details displayed in the **File → Preferences** dialog window? Additionally, it would be helpful if you could attach the complete error message from the **Program Output** window.

Best, 
Abiy

**Reply by Mewtwo1124** [reply]
Hi Abiy,
           I have attached my setup details and complete error message from the Program Output window. Please let me know how to proceed.
![Screenshot 2025-01-15 194130](https://github.com/user-attachments/assets/8996ab55-9eba-4b44-b213-582b3a286618)
![Screenshot 2025-01-15 194105](https://github.com/user-attachments/assets/e108a3c8-ca3f-4e83-ad43-661e12cae490)
![Screenshot 2025-01-15 194040](https://github.com/user-attachments/assets/3e2c8fa4-e827-4c04-bcc6-14056c3c804d)
![Screenshot 2025-01-15 193232](https://github.com/user-attachments/assets/335e81a1-ac0f-4ea9-812d-a920f29d37f5)

Thanks
Tanmay

**Comment by abiyfantaye** [comment]
Hi Tanmay, 

Thank you for providing more information. Can you please make sure that the path to Python points to "{YOUR_DOWNLOAD_DIR}/WE-UQ_Windows_Download/applications/python/python.exe" 

Also, check if the "BackendApplications" points to  "{YOUR_DOWNLOAD_DIR}/WE-UQ_Windows_Download/applications".  You can do so by checking the box next to "BackendApplications". 

I can't fully see the "Python" and "BackendApplication" paths. I see two "WE-UQ_Windows_Download" in backend applications but not in Python. Can you check if both paths exist, test it one more time, and let me know?  

Best, 
Abiy

**Reply by Mewtwo1124** [reply]
Hi Abiy,
       Thanks for the reply. The application is working now. I have got two questions:
1.  I tried installing and compiling the Tinf tool but the compilation gives errors. I tried to compile by Building the UI from Source and by Compiling the Source Code in OpenFOAM. I tried compiling OpenFOAM10 and OpenFOAM11 codes using the respective OpenFOAM versions. I have attached the errors to this reply for your reference. Please let me know how to fix them.
This is the error I get when I try to compile the OpenFOAM10 code. The error goes on for long.

![image](https://github.com/user-attachments/assets/008a1b84-225c-49a3-931e-9ef754ac49b1)
![image](https://github.com/user-attachments/assets/07adb60c-6bdb-4301-8158-d0c890ea9957)
![image](https://github.com/user-attachments/assets/17f13b0c-8aa6-4e68-b836-21a184ab58a5)

This is the error I get when I try to compile from the source.
![image](https://github.com/user-attachments/assets/92cf5fc6-e12d-4b43-85e5-0d283699ac65)
![image](https://github.com/user-attachments/assets/8e28225c-9c2b-414b-b0bc-605e609e3b29)
![image](https://github.com/user-attachments/assets/3602f80b-62d3-438c-b668-c034997de19d)

This is the error I get when I try to compile the OpenFOAM11 code
![image](https://github.com/user-attachments/assets/2938c986-6e64-4830-ba64-39b02b4ae94a)


2.  If I want to perform my own simulation in WE_UQ, do I need to create a separate JSON file or are the OpenFOAM case files enough to run the simulations?


Thanks
Tanmay

**Reply by abiyfantaye** [reply]
Hi Tanmay, 

I am glad to hear that WE-UQ is finally working for you. I'm looking into the issue related to TInf compilation and will get back to you soon. 

Best, 
Abiy

**Reply by Mewtwo1124** [reply]
Hi Abiy,
          Thanks for getting back to me. If I want to perform a simulation on a domain that is not in the ones provided in the examples then how can I do that?

Thanks
Tanmay
